src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -74,10 +74,25 @@ public override void WriteUsingDirective(CodeRenderingContext context, UsingDire
 
         if (node.Source is { FilePath: not null } sourceSpan)
         {
+            // Write in the form of
+            // using
+            // #line ...
+            // Content
+            //
+            // This makes it so the spacing is avoided for any formatting that happens on the generated code.
+            context.CodeWriter.WriteLine("using");
             using (context.CodeWriter.BuildLinePragma(sourceSpan, context, suppressLineDefaultAndHidden: !node.AppendLineDefaultAndHidden))
             {
-                context.AddSourceMappingFor(node);
-                context.CodeWriter.WriteUsing(node.Content);
+                var usingLength = "using ".Length;
```

Comment:
Should this find the actual position of the first non-whitespace character after `using`?

In other words, what happens if the user has `@using           System;`

Reply:
ah that's a good point. I can update and add a test for that

Reply:
So, you have identified a pre-existing bug with source mapping. We already get the sourcemapping completely wrong in that case. It currently would get written as `using System;` and ignore the spaces. That's because the `node.Content` doesn't have the spaces in it, they're trimmed out.